### DISCUSSION TITLE
Proveedores de servicios externos

### DISCUSSION THREAD

**Discussion by Zoomelectrico** [Avila-Tek/fullstack-template#7] [Ideas]
# Proveedores de servicios externos

Me gustaría abrir una discusión para conversar acerca de los servicios de terceros que estamos utilizando y si podemos utilizar un mejor proveedor para ese servicio. La dinámica será abrir un nuevo hilo de conversación por cada proveedor servicio de tercero que usemos o que tengamos que usar para definir la mejor alternativa posible.

**Comment by Zoomelectrico** [comment]
# 📩 Proveedores de email

En este momento estamos utilizando [postmark](https://postmarkapp.com) para todos nuestro proyectos. Me gustaría leer las experiencias que hemos tenido con este proveedor y ver alternativas, por ejemplo:

| Proveedor                                   | Precio  | Nota sobre el precio | Notas                                                                                                            | Proyecto |
| :------------------------------------------ | :------ | :------------------- | :--------------------------------------------------------------------------------------------------------------- | :------- |
| [Postmark](https://postmarkapp.com/)        | 55 $/mo | Por 50k correos      | Es nuestro estandar                                                                                              | todos    |
| [Resend](https://resend.com/)               | 20 $/mo | Por 50K correos      | Tienen [react.email](https://react.email/docs/components/tailwind#tailwind-configuration-with-px-instead-of-rem) | N/A      |
| [Sendgrid](https://sendgrid.com/en-us)      | 20 $/mo | Por 50k correos      | Tienen problemas con las ips compartidas                                                                         | armi     |
| [[AWS SES](https://aws.amazon.com/es/ses/)] | 6 $/mo  | Por 50K correos      | Requieren de mucho trabajo para las plantillas                                                                   | dinbog   |

**Reply by Leopgf** [reply]
Pregunta 1: Qué criterios vamos a tomar para escoger el proveedor de mail que mejor se adapte a nuestros requerimientos?
Propuesta de criterios: Precio, facilidad para integrar, confiabilidad de envío del correo, flexibilidad de configuraciones. 
Propuesta de criterios 2: Si definimos unos criterios para tomar la decisión, deberíamos asignarle una ponderación a cada criterio

**Comment by Zoomelectrico** [comment]
# 🔍 Proveedores de búsqueda

En este momento estamos utilizando [algolia](https://algolia.com/) para todos nuestro proyectos. Me gustaría leer las
experiencias que hemos tenido con este proveedor y ver alternativas, por ejemplo:

<table>
  <thead>
    <tr>
      <th>Proveedor</th>
      <th>Precio</th>
      <th>Nota sobre el precio</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><a href="https://www.algolia.com/es/pricing/">Algolia</a></td>
      <td>Pay-As-You-Go</td>
      <td>
        <ul>
          <li>$0.50 per additional 1k search requests</li>
          <li>$0.40 per additional 1k records/month</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td><a href="https://typesense.org">Typesense</a></td>
      <td>Self-Hosted / Cloud option</td>
      <td>
        <ul>
          <li>Self hosted (up to us)</li>
          <li>Cloud min 21.60 USD/mo</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td><a href="https://www.elastic.co/es/elasticsearch">Elasticsearch</a></td>
      <td>Cloud offering</td>
      <td>
        <ul>
          <li>$120/mo min</li>
        </ul>
      </td>
    </tr>
     <tr>
      <td><a href="https://www.meilisearch.com/pricing">Meilisearch</a></td>
      <td>$300/mo</td>
      <td>
        <ul>
          <li>Plan de 30 $/mo</li>
          <li>Self hosted</li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


Para ser honesto, creo que nuestra experiencia con algolia ha sido buena, ya la curva de aprendizaje de la herramienta la tuvimos y creo valioso, entender si por el costo de algolia vale la pena mantenernos, estuve buscando articulos al respecto y me encontre con los siguientes

1. [https://typesense.org/typesense-vs-algolia-vs-elasticsearch-vs-meilisearch/](https://typesense.org/typesense-vs-algolia-vs-elasticsearch-vs-meilisearch/)
2. [https://www.luigisbox.com/typesense-vs-algolia/](https://www.luigisbox.com/typesense-vs-algolia/)
3. [https://blog.tarkalabs.com/searching-for-the-perfect-fit-our-move-from-algolia-to-typesense-d8ee3eef48e3](https://blog.tarkalabs.com/searching-for-the-perfect-fit-our-move-from-algolia-to-typesense-d8ee3eef48e3)

**Comment by Zoomelectrico** [comment]
# 💳 Proveedores de pago

| Proveedor                               | Proyectos |
| :-------------------------------------- | :-------- |
| [Stripe](https://stripe.com/es-us)      | Spacars   |
| [Luka](https://joinluka.com/)           | Spacars   |
| [Mi Banco](https://www.mibanco.com.ve/) | Estei     |
| [Bancamiga](https://www.bancamiga.com/) | Spacars   |
| [BNC](https://www.bncenlinea.com/)      | Zas!      |
| [Instapagos](https://instapago.com/)    | Meit!     |

**Reply by lesanpi** [reply]
Alternativas de Proveedores de Pago:
* Banco Mercantil, que ofrece:
Pago Móvil C2P
Pago con TDC
Pago con TDD
Búsqueda de Transferencias

**Comment by Zoomelectrico** [comment]
# 🔒 Servicios de autenticación

| Servicios                                                                          | Proyectos         |
| :--------------------------------------------------------------------------------- | :---------------- |
| [next-auth](https://authjs.dev/)                                                   | La mayoría        |
| [Keycloack](https://www.keycloak.org/)                                             | Cines U           |
| [Firebase](https://firebase.google.com/docs/auth?hl=es-419)                        | Socado            |
| [Azure B2C](https://learn.microsoft.com/es-es/azure/active-directory-b2c/overview) | Mercantil Seguros |
| [Supertoken](https://supertokens.com/)                                             | N/A               |
| [Supabase](https://supabase.com/)                                                  | N/A               |

**Reply by lesanpi** [reply]
Alternativas de Servicios de Autenticación:
* Amplify AWS.
* Passport

**Reply by Leopgf** [reply]
| Servicios  | Proyectos |
| ------------- | ------------- |
| [Clerk](https://clerk.com)  | Armi  |

**Reply by jesussoares** [reply]
Alternativa de servicio de autenticación:
- [Auth.js](https://authjs.dev/)
